### PR TITLE
bugfix/20968 synchronization of connector values

### PR DESCRIPTION
**Pull request by jomar-honsi**
Adding invisible column with row keys (unique identifiers).

### CODE REVIEW

**Comment by jomar-honsi** [comment]
This is a renaming of PR and branch as requested by @vazonik . Original [PR](https://github.com/highcharts/highcharts/pull/21242#issue-2322853579).

**Comment by jomar-honsi** [comment]
Renaming of PR and branch as requestet by @vazonik.
Original [PR](https://github.com/highcharts/highcharts/pull/21242#issue-2322853579).

**Comment by jomar-honsi** [comment]
Test.

File: ts/Data/DataTable.ts [hunk 1]
```diff
@@ -191,9 +191,10 @@ class DataTable implements DataEvent.Emitter {
         this.modified = this;
         this.rowCount = 0;
         this.versionTag = uniqueKey();
+        this.rowKeysId = options.rowKeysId;
 
-        const columns = options.columns || {},
-            columnNames = Object.keys(columns),
+        const columns = options.columns || {};
+        const columnNames = Object.keys(columns),
             thisColumns = this.columns;
```
Comment (vazonik) [inline ts/Data/DataTable.ts:198]: Use consistently
```ts
const a = ...,
   b = ...,
   c = ...;
```
or
```ts
const a = ...;
const b = ...;
const c = ...;
```

**Review by vazonik — CHANGES_REQUESTED** [review]
I also noticed that there are problems with Highlight Sync. Some rows are skipped and the next ones highlight incorrect points, if there is a shift equal to the number of previously skipped points. Could you investigate it? It seems that the original row index should be used somewhere in the syncs.

**Comment by jomar-honsi** [comment]
> I also noticed that there are problems with Highlight Sync. Some rows are skipped and the next ones highlight incorrect points, if there is a shift equal to the number of previously skipped points. Could you investigate it? It seems that the original row index should be used somewhere in the syncs.

This issue is reported, I'd prefer to deal with it separately: #21296.

**Review by vazonik — APPROVED** [review]
> This issue is reported, I'd prefer to deal with it separately: https://github.com/highcharts/highcharts/issues/21296.

Okay, that makes sense!

**Review by sebastianbochan — APPROVED** [review]

**Pull request merged**